hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/index/bloom/SparkHoodieBloomIndexHelper.java

```diff
@@ -75,7 +80,9 @@ public static SparkHoodieBloomIndexHelper getInstance() {
         + config.getBloomIndexParallelism() + "}");
 
     JavaRDD<List<HoodieKeyLookupResult>> keyLookupResultRDD;
-    if (config.isMetadataBloomFilterIndexEnabled()) {
+    if (config.getBloomIndexUseMetadata()
+        && getCompletedMetadataPartitions(hoodieTable.getMetaClient().getTableConfig())
```

Comment:
can we add an api to tableConfig only getCompletedMetadataPartitions(). we can compute once and return the same rather than parsing the table config entry again and again.

Reply:
`HoodieTableMetadataUtil::getCompletedMetadataPartitions` is an existing API.  We can tackle this code cleanup in a separate PR: HUDI-3836.